### PR TITLE
Prevent PHP hang when subclass extends wrong class

### PR DESCRIPTION
**Pull request by darron1217**
Issue #66

### CODE REVIEW

**Comment by jonspalmer** [comment]
Nice work. I'm sure this works but can we add the test case that covers it?

**Comment by darron1217** [comment]
I'm not good at writing test, but I'll give it a try

**Comment by darron1217** [comment]
I have some problem while testing, I did `composer install` and `phpunit` but it emits `Illuminate\Database\QueryException : could not find driver (SQL: select * from sqlite_master where type = 'table' and name = migrations)`

How can I fix it?

**Comment by darron1217** [comment]
Never mind, I didn't add `extension=pdo_sqlite` to my php.ini

**Comment by darron1217** [comment]
@jonspalmer 
I've added a test. Please check

File: tests/Fixtures/AudioMP3.php [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Nanigans\SingleTableInheritance\Tests\Fixtures;
+
+class AudioMP3 extends File {
+
```
Comment (jonspalmer) [inline tests/Fixtures/AudioMP3.php:6]: Can we add a comment here as to what is broken here?

File: src/SingleTableInheritanceTrait.php [hunk 2]
```diff
@@ -63,6 +64,9 @@ public static function getSingleTableTypeMap() {
       // prevent infinite recursion if the singleTableSubclasses is inherited
       if (!in_array($calledClass, $subclasses)) {
         foreach ($subclasses as $subclass) {
+          if(!is_subclass_of($subclass, $calledClass)) {
+            throw new SingleTableInheritanceWrongInheritanceException('Subclass must extend its parent class. Parent: '.$calledClass.' | Subclass: '.$subclass);
+          }
```
Comment (jonspalmer) [inline src/SingleTableInheritanceTrait.php:69]: a) Should SingleTableInheritanceWrongInheritanceException take the two classes and generate the message itself?
b) let's use String interpolation vs concatenation

Comment (darron1217) [reply src/SingleTableInheritanceTrait.php:69]: a) Can you provide an example for me? I need your help.
b) Got it.

This is my best for now.
```php
          if(!is_subclass_of($subclass, $calledClass)) {
            throw new SingleTableInheritanceWrongInheritanceException("Subclass must extend its parent class. Parent: $calledClass | Subclass: $subclass");
          }
```

**Comment by darron1217** [comment]
@jonspalmer I've modified my PR. Please check

**Comment by darron1217** [comment]
@jonspalmer ping

**Comment by jonspalmer** [comment]
Thanks for the contribution sorry it took me a while to circle back on this one.

**Pull request merged**